server/container_create.go

```diff
@@ -798,19 +807,55 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 	specgen.HostSpecific = true
 	specgen.ClearProcessRlimits()
 
-	var readOnlyRootfs bool
+	readOnlyRootfs := s.config.ReadOnly
+
 	var privileged bool
 	if containerConfig.GetLinux().GetSecurityContext() != nil {
 		if containerConfig.GetLinux().GetSecurityContext().Privileged {
 			privileged = true
 		}
-
 		if containerConfig.GetLinux().GetSecurityContext().ReadonlyRootfs {
 			readOnlyRootfs = true
-			specgen.SetRootReadonly(true)
 		}
 	}
+	specgen.SetRootReadonly(readOnlyRootfs)
 
+	if s.config.ReadOnly {
```

Comment:
Why is this independent of `containerConfig.GetLinux().GetSecurityContext().ReadonlyRootfs`?  I'd expect to switch on `readOnlyRootfs`.

Reply:
Well the goal of this patch was to only do the mounts if the CRI-O was setup to run containers read/only.  If the user (kubernetes) specifies readonly we don't currently do anything with /run, /tmp and /var/tmp.  I am not sure we want to change that default behaviour, unless the admin setup CRI-O in that mode.

@mrunalp WDYT

Reply:
@rhatdan do we want to run in read only if the user says read-only, but cri-o is not running with a read-only flag?

Reply:
nvm, just saw your comment above.

Reply:
Yeah, I don't think we should modify the CRI expected behavior for the flag. This is more a CRI-O optional feature.